File: DAQ_Zynq_GUI/SW/Portal/app/dac_jmp_backend.jl

function set_cfg_py(
    t_pump,
    t_probe,
    f_2larmor,
    V_pump1,
    V_pump2,
    V_probe
)
    println("SET_CFG_PY ENTER")

    dac = DAC_Jmp(get_portal(), PG_DAC_JMP)

    try
        set_cfg!(
            dac,
            t_pump=t_pump,
            t_probe=t_probe,
            f_2larmor=f_2larmor,
            V_pump1=V_pump1,
            V_pump2=V_pump2,
            V_probe=V_probe
        )
    catch e
        println("SET_CFG_PY ERROR: $e")
        rethrow()   
    end

    println("SET_CFG_PY EXIT")
end

function mon_probe_py()

    dac = DAC_Jmp(get_portal(), PG_DAC_JMP)

    try
        while probe(dac);  sleep(0.0005); end   # wait FE
        while !probe(dac); sleep(0.0005); end   # wait RE
        t_probe_start = time()
        while probe(dac);  sleep(0.0005); end   # wait FE
        t_probe_end = time()

		t_probe_ms = (t_probe_end - t_probe_start)*1e3

		println("T_PROBE_MS = $t_probe_ms")
    catch e
        println("MON_PROBE_PY ERROR: $e")
    end
end

File: DAQ_Zynq_GUI/SW/Portal/app/dac_jmp_backend.py
from juliacall import Main as jl
import os

THIS_DIR = os.path.dirname(os.path.abspath(__file__))

from juliacall import Main as jl

import numpy as np

jl.include(os.path.join(THIS_DIR, "dac_jmp_backend.jl"))


def set_cfg(
        t_pump,
        t_probe,
        f_2larmor,
        V_pump1,
        V_pump2,
        V_probe):
    
    print("===== DAC_JMP CONFIG =====")
    print("t_pump     =", t_pump)
    print("t_probe    =", t_probe)
    print("f_2larmor  =", f_2larmor)
    print("V_pump1    =", V_pump1)
    print("V_pump2    =", V_pump2)
    print("V_probe    =", V_probe)

    jl.set_cfg_py(
        t_pump,
        t_probe,
        f_2larmor,
        V_pump1,
        V_pump2,
        V_probe
    )


def probe():
    return bool(jl.probe_py())